### PR TITLE
GDAL: add v3.10.0

### PR DESCRIPTION
**Pull request by adamjstewart**
https://github.com/OSGeo/gdal/releases/tag/v3.10.0

### CODE REVIEW

File: var/spack/repos/builtin/packages/gdal/package.py
Comment (wdconinc) [inline var/spack/repos/builtin/packages/gdal/package.py:360]: Should this PR be dependent on another PR that adds `libopendrive`, or will GDAL use a vendored copy if no version is found?

**Review by wdconinc — COMMENTED** [review]
Checksum verified.

File: var/spack/repos/builtin/packages/gdal/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/gdal/package.py:360]: No vendored copy. GDAL is an I/O library with ~200 drivers, each of which requires an external software dependency. Many but not all of those are already packaged in Spack. Technically, all of these commented out dependencies should also have a corresponding:
```python
conflicts("+opendrive")
```
to prevent the use of that variant until someone adds a package, I just haven't been bothering to do that. Let me know if you want me to add conflicts (or comment out variants).

**Review by wdconinc — APPROVED** [review]
Thanks for the clarification on `opendrive`. A conflict would prevent someone from using `opendrive` provided as an external. I'll defer to you as maintainer on this.

**Pull request merged**
